examples/connext_dds/group_coherent_presentation/c++11/group_coherent_publisher.cxx

```diff
@@ -0,0 +1,150 @@
+/*
+* (c) Copyright, Real-Time Innovations, 2021.  All rights reserved.
+* RTI grants Licensee a license to use, modify, compile, and create derivative
+* works of the software solely for use with RTI Connext DDS. Licensee may
+* redistribute copies of the software provided that all such copies are subject
+* to this license. The software is provided "as is", with no warranty of any
+* type, including any warranty for fitness for any purpose. RTI is under no
+* obligation to maintain or support the software. RTI shall not be liable for
+* any incidental or consequential damages arising out of the use or inability
+* to use the software.
+*/
+
+#include <iostream>
+
+#include <dds/pub/ddspub.hpp>
+#include <rti/util/util.hpp>      
+#include <rti/config/Logger.hpp>
+
+#include "application.hpp"
+#include "group_coherent.hpp"
+
+using namespace dds::core;
+using namespace dds::core::policy;
+using namespace dds::pub::qos;
+
+void run_publisher_application(unsigned int domain_id, unsigned int set_count)
+{
+    // Start communicating in a domain, usually one participant per application
+    dds::domain::DomainParticipant participant(domain_id);
+
+    // Create two Topics with names and a datatypes
+    dds::topic::Topic<CheckpointTime> time_topic(
+            participant, 
+            "Checkpoint Time");
+    dds::topic::Topic<CheckpointPlace> place_topic(
+            participant, 
+            "Checkpoint Place");
+
+    // Retrieve the Publisher QoS, from USER_QOS_PROFILES.xml.
+    PublisherQos publisher_qos = QosProvider::Default().publisher_qos();
+
+    // If you want to change the Publisher's QoS programmatically rather
+    // than using the XML file, you will need to uncomment the following line.
+    // publisher_qos << Presentation::GroupAccessScope(true, true);
```

Comment:
Instead of leaving code commented out, I suggest adding an argument to the run_publisher_function, `bool use_xml_qos = true`, and then conditionally get the Qos from XML or programmatically.